### PR TITLE
Atoms as islands

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Here we're moving a bunch of atoms into Islands. Specifically,

```
	QandaAtom
	GuideAtom
	ProfileAtom
	TimelineAtom
	ChartAtom
	PersonalityQuizAtom
	KnowledgeQuizAtom
```

We're also add version `22.2.0` of `@guardian/atoms-rendering` which allows us to remove the tracking handlers

## Why?
Islands are cool.

Why are we removing the tracking handlers? Because adding these on the server and then trying to hydrate them was non intuitive and added complexity. By adding a default to the atom itself we can simply not include the handler and use the fallback instead.

## What are these wrapper components?
For code that's imported from an external lib it's not possible for our `Island` abstraction to dynamcally import it because we currently require island components to exist in `/component`. To get around this we use a simple wrapper, as shown here.

### CODE REVIEW

File: dotcom-rendering/src/web/components/App.tsx [hunk 1]
```diff
@@ -11,16 +11,6 @@ import { GetMatchNav } from '@frontend/web/components/GetMatchNav';
...
 
-import {
-	QandaAtom,
-	GuideAtom,
-	ProfileAtom,
-	TimelineAtom,
-	ChartAtom,
-	PersonalityQuizAtom,
-	KnowledgeQuizAtom,
-} from '@guardian/atoms-rendering';
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/components/App.tsx:22]: If I'm reading this right, does this mean we were importing all these atoms on the page no matter what?! Glad to see it go!

File: dotcom-rendering/src/web/components/App.tsx [hunk 2]
```diff
@@ -11,16 +11,6 @@ import { GetMatchNav } from '@frontend/web/components/GetMatchNav';
...
-
 import { AudioAtomWrapper } from '@frontend/web/components/AudioAtomWrapper';
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/components/App.tsx:14]: What's happening to the audio atom? Will this be handled by a separate PR?

File: dotcom-rendering/src/web/components/App.tsx (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/App.tsx:22]: Yes! 😱

File: dotcom-rendering/src/web/components/App.tsx (hunk 2 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/App.tsx:14]: Good spot. I've added a ticket for this one.

File: dotcom-rendering/src/web/components/App.tsx [hunk 3]
```diff
@@ -525,113 +437,6 @@ export const App = ({ CAPI, ophanRecord }: Props) => {
...
-							'DISLIKE',
-						)}
-						expandCallback={componentEventHandler(
-							'QANDA_ATOM',
-							qandaAtom.id,
-							'EXPAND',
-						)}
-					/>
-				</HydrateOnce>
-			))}
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/components/App.tsx:554]: I see that for Qanda atoms (no idea what these are) we are no longer reporting ophan data. Is this perhaps something we can do in the wrapper to make sure we're not loosing functionality?

Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/App.tsx:554]: The tracking requests still get sent. It's just I made these tracking props optional and then added a default inside the atoms themselves. 

https://github.com/guardian/atoms-rendering/pull/325

Comment (OllysCoding) [reply dotcom-rendering/src/web/components/App.tsx:554]: 👍 Thanks for clarifying!

**Review by OllysCoding — APPROVED** [review]
+1 Looks good!
